prov/tcp/src/tcpx.h

```diff
@@ -188,9 +188,10 @@ struct tcpx_pep {
 struct tcpx_ep {
 	struct util_ep		util_ep;
 	SOCKET			conn_fd;
-	struct dlist_entry	ep_entry;
 	struct dlist_entry	rx_queue;
 	struct dlist_entry	tx_queue;
+	/* lock for accessing rx & tx queues */
+	fastlock_t		queue_lock;
```

Comment:
The udp provider ends up using the tx/rx cq locks when sending or receiving.  This avoids adding extra locks when a completion must also be written.  Please see if a similar model would be useful here.

Reply:
I think it is good idea to use the cq locks.  That will decrease the number of locks for sure. But I am wondering if it would add considerable performance overhead.